pkg/controllers/clusterschedulingpolicysnapshot/controller.go

```diff
@@ -33,48 +32,58 @@ import (
 type Reconciler struct {
 	client.Client
 
 	// PlacementController exposes the placement queue for the reconciler to push to.
 	PlacementController controller.Controller
 }
 
-// Reconcile triggers a single CRP reconcile round when scheduling policy has changed.
+// Reconcile triggers a single placement reconcile round when scheduling policy has changed.
 func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	name := req.NamespacedName
-	snapshot := fleetv1beta1.ClusterSchedulingPolicySnapshot{}
-	snapshotKRef := klog.KRef(name.Namespace, name.Name)
+	var snapshot fleetv1beta1.PolicySnapshotObj
+	var err error
+	snapshotKRef := klog.KRef(req.Namespace, req.Name)
 
 	startTime := time.Now()
-	klog.V(2).InfoS("Reconciliation starts", "clusterSchedulingPolicySnapshot", snapshotKRef)
+	klog.V(2).InfoS("Reconciliation starts", "policySnapshot", snapshotKRef)
 	defer func() {
 		latency := time.Since(startTime).Milliseconds()
-		klog.V(2).InfoS("Reconciliation ends", "clusterSchedulingPolicySnapshot", snapshotKRef, "latency", latency)
+		klog.V(2).InfoS("Reconciliation ends", "policySnapshot", snapshotKRef, "latency", latency)
 	}()
 
-	if err := r.Client.Get(ctx, name, &snapshot); err != nil {
-		if errors.IsNotFound(err) {
-			klog.V(4).InfoS("Ignoring NotFound clusterSchedulingPolicySnapshot", "clusterSchedulingPolicySnapshot", snapshotKRef)
-			return ctrl.Result{}, nil
+	if req.Namespace == "" {
+		// ClusterSchedulingPolicySnapshot (cluster-scoped)
+		var clusterSchedulingPolicySnapshot fleetv1beta1.ClusterSchedulingPolicySnapshot
+		if err = r.Client.Get(ctx, req.NamespacedName, &clusterSchedulingPolicySnapshot); err != nil {
+			klog.ErrorS(err, "Failed to get cluster scheduling policy snapshot", "policySnapshot", snapshotKRef)
+			return ctrl.Result{}, controller.NewAPIServerError(true, client.IgnoreNotFound(err))
 		}
-		klog.ErrorS(err, "Failed to get clusterSchedulingPolicySnapshot", "clusterSchedulingPolicySnapshot", snapshotKRef)
-		return ctrl.Result{}, controller.NewAPIServerError(true, err)
+		snapshot = &clusterSchedulingPolicySnapshot
+	} else {
+		// schedulingPolicySnapshot (namespaced)
+		var schedulingPolicySnapshot fleetv1beta1.SchedulingPolicySnapshot
+		if err = r.Client.Get(ctx, req.NamespacedName, &schedulingPolicySnapshot); err != nil {
+			klog.ErrorS(err, "Failed to get scheduling policy snapshot", "policySnapshot", snapshotKRef)
+			return ctrl.Result{}, controller.NewAPIServerError(true, client.IgnoreNotFound(err))
+		}
+		snapshot = &schedulingPolicySnapshot
 	}
-	crp := snapshot.Labels[fleetv1beta1.PlacementTrackingLabel]
-	if len(crp) == 0 {
-		err := fmt.Errorf("invalid label value %s", fleetv1beta1.PlacementTrackingLabel)
-		klog.ErrorS(err, "Invalid clusterSchedulingPolicySnapshot", "clusterSchedulingPolicySnapshot", snapshotKRef)
+
+	placementName, exist := snapshot.GetLabels()[fleetv1beta1.PlacementTrackingLabel]
```

Comment:
nit, no need to add the exist.

if not exist, the placementName will be empty string.